load-testing/tests/relays_stress_helpers_test.go

```diff
@@ -75,28 +81,46 @@ type actorLoadTestIncrementPlan struct {
 	maxActorCount int64
 }
 
-// setupTxEventListeners sets up the transaction event listeners to observe the
-// transactions committed on-chain.
-func (s *relaysSuite) setupTxEventListeners() {
-	eventsQueryClient := testeventsquery.NewLocalnetClient(s.TestingT.(*testing.T))
+// setupEventListeners sets up the event listeners for the relays suite.
+// It listens to both tx and block events to keep track of the events that are happening
+// on the chain.
+func (s *relaysSuite) setupEventListeners(pocketNode string) {
+	// Set up the blockClient that will be notifying the suite about the committed blocks.
+	eventsObs, eventsObsCh := channel.NewObservable[[]types.Event]()
+	s.eventsObs = eventsObs
 
-	deps := depinject.Supply(eventsQueryClient)
-	eventsReplayClient, err := events.NewEventsReplayClient(
+	s.blockClient = testblock.NewLocalnetClient(s.ctx, s.TestingT.(*testing.T))
+	channel.ForEach(
 		s.ctx,
-		deps,
-		newTxEventSubscriptionQuery,
-		tx.UnmarshalTxResult,
-		eventsReplayClientBufferSize,
-	)
-	require.NoError(s, err)
+		s.blockClient.CommittedBlocksSequence(s.ctx),
+		func(ctx context.Context, block client.Block) {
```

Comment:
NIT: Can you assign this to a locally scope function so the `.ForEach` is easier to read?